packages/dev/core/src/Meshes/csg2.ts

```diff
@@ -0,0 +1,513 @@
+     * @returns a new csg
+     */
+    public subtract(csg: CSG2) {
+        return this._process("difference", csg);
+    }
+
+    /**
+     * Run an intersection operation between two CSG
+     * @param csg defines the CSG to use to create the intersection
+     * @returns a new csg
+     */
+    public intersect(csg: CSG2) {
+        return this._process("intersection", csg);
+    }
+
+    /**
+     * Run an union operation between two CSG
+     * @param csg defines the CSG to use to create the union
+     * @returns a new csg
+     */
+    public add(csg: CSG2) {
+        return this._process("union", csg);
+    }
+
+    /**
+     * Print debug information about the CSG
+     */
+    public printDebug() {
+        Logger.Log("Genus:" + this._manifold.genus());
+        const properties = this._manifold.getProperties();
+        Logger.Log("Volume:" + properties.volume);
+        Logger.Log("surface area:" + properties.surfaceArea);
+    }
+
+    /**
+     * Generate a vertex data from the CSG
+     * @param options defines the options to use to rebuild the vertex data
+     * @returns a new vertex data
+     */
+    public toVertexData(options?: Partial<IVertexDataRebuildOptions>): VertexData {
+        const localOptions = {
+            rebuildNormals: false,
+            ...options,
+        };
+        const vertexData = new VertexData();
+        const normalComponent = this._vertexStructure.find((c) => c.kind === VertexBuffer.NormalKind);
+        const manifoldMesh: IManifoldMesh = this._manifold.getMesh(localOptions.rebuildNormals && normalComponent ? [3, 4, 5] : undefined);
+
+        vertexData.indices = new Uint16Array(manifoldMesh.triVerts);
+
+        for (let i = 0; i < manifoldMesh.triVerts.length; i += 3) {
+            vertexData.indices[i] = manifoldMesh.triVerts[i + 2];
+            vertexData.indices[i + 1] = manifoldMesh.triVerts[i + 1];
+            vertexData.indices[i + 2] = manifoldMesh.triVerts[i];
+        }
+
+        const vertexCount = manifoldMesh.vertProperties.length / manifoldMesh.numProp;
+
+        // Positions (first one is always position)
+        const positions = new Float32Array(vertexCount * this._vertexStructure[0].stride);
+        for (let i = 0; i < vertexCount; i++) {
+            positions[i * 3] = manifoldMesh.vertProperties[i * manifoldMesh.numProp];
+            positions[i * 3 + 1] = manifoldMesh.vertProperties[i * manifoldMesh.numProp + 1];
+            positions[i * 3 + 2] = manifoldMesh.vertProperties[i * manifoldMesh.numProp + 2];
+        }
+        vertexData.positions = positions;
+
+        // Other attributes
+        let offset = 3;
+        for (let componentIndex = 1; componentIndex < this._vertexStructure.length; componentIndex++) {
+            const component = this._vertexStructure[componentIndex];
+            // Normals (special case as we need to convert the data)
+            if (component.kind === VertexBuffer.NormalKind) {

```

Comment:
I'm not sure to see the difference with the other code path?

Reply:
nice catch! I forgot to revert back!